debezium-core/src/main/java/io/debezium/relational/Key.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.annotation.Immutable;
+import io.debezium.function.Predicates;
+
+/**
+ * An immutable definition of the Table.Key. By default the key will be compound of the primary key column. <br>
+ * Key can be customized with a {@code CustomKeyMapper} 
+ * 
+ * @author Guillaume Rosauro
+ *
+ */
+@Immutable
+public class Key {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(Key.class);
+    
+    private final Table table;
+    private final KeyMapper keyMapper;
+    
+    
+    private Key(Table table, KeyMapper keyMapper) {
+        this.table = table;
+        this.keyMapper = keyMapper;
+    }
+
+    /**
+     * 
+     * @return the columns {@link Column} taking part of the key.
+     */
+    public List<Column> keyColumns() {
+        return keyMapper.matches(table);
+    }
+    
+    /**
```

Comment:
This comment should be removed, it doesn't add value really.

Reply:
comment removed